### ISSUE TITLE
`v-for` looses types for iterating over object

### ISSUE THREAD

**Issue by zigomir** [johnsoncodehk/volar#117]
Hi – your plugin is amazing, thanks so much for creating it!

This issue is probably similar to https://github.com/johnsoncodehk/volar/issues/104 but maybe this can be improved.

For a component

```vue
<template>
  <div>
    <div v-for="(value, key) in obj">
      <span>{{key}}</span>: {{ value.toFixed() }}
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  setup() {
    const obj: Record<'key1' | 'key2', number> = {
      key1: 1,
      key2: 2,
    }

    return { obj }
  }
})
</script>
```

I get these on hover

1. `obj` is correctly typed
<img width="551" alt="Screenshot 2021-03-29 at 15 13 19" src="https://user-images.githubusercontent.com/957321/112842525-eb7e7600-90a1-11eb-8505-a9ffb2cf1589.png">

2. `value` is `any` while it should be `number`
<img width="287" alt="Screenshot 2021-03-29 at 15 13 24" src="https://user-images.githubusercontent.com/957321/112842570-f89b6500-90a1-11eb-84bf-6cfb20426221.png">

3. `key` is `string` while it should be `'key1' | 'key2'`
<img width="179" alt="Screenshot 2021-03-29 at 15 14 16" src="https://user-images.githubusercontent.com/957321/112842598-0224cd00-90a2-11eb-853b-d4feed4483f2.png">

This is a [reproduction repo](https://github.com/zigomir/vue-tsc-v-for).

**Comment by johnsoncodehk** [comment]
Fixed in 0.22.24, thanks for the report!

**Comment by zigomir** [comment]
Amazing – thank you very much – it now works very well with a lot more complex object! 🎉
